src/sentry/static/sentry/app/components/shareIssue.tsx

```diff
@@ -0,0 +1,242 @@
+import DropdownLink from 'app/components/dropdownLink';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import Switch from 'app/components/switch';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+
+type ContainerProps = {
+  shareUrl: string;
+  onConfirming: () => void;
+  onConfirm: () => void;
+  onCancel: () => void;
+};
+
+type UrlRef = AutoSelectText | null;
+
+class ShareUrlContainer extends React.Component<ContainerProps> {
+  urlRef?: UrlRef;
+
+  // Select URL when its container is clicked
+  handleCopyClick = () => {
+    this.urlRef?.selectText();
+  };
+
+  handleUrlMount = (ref: UrlRef) => {
+    this.urlRef = ref;
+    // Always select url if it's available
+    this.urlRef?.selectText();
+  };
+
+  render() {
+    const {shareUrl, onConfirming, onCancel, onConfirm} = this.props;
+
+    return (
+      <UrlContainer>
+        <StyledTextContainer>
+          <StyledAutoSelectText ref={ref => this.handleUrlMount(ref)}>
+            {shareUrl}
+          </StyledAutoSelectText>
+        </StyledTextContainer>
+
+        <Clipboard hideUnsupported value={shareUrl}>
+          <ClipboardButton
+            title={t('Copy to clipboard')}
+            borderless
+            size="xsmall"
+            onClick={this.handleCopyClick}
+            icon={<IconCopy />}
+          />
+        </Clipboard>
+
+        <Confirm
+          message={t(
+            'You are about to regenerate a new shared URL. Your previously shared URL will no longer work. Do you want to continue?'
+          )}
+          onCancel={onCancel}
+          onConfirming={onConfirming}
+          onConfirm={onConfirm}
+        >
+          <ReshareButton
+            title={t('Generate new URL')}
+            borderless
+            size="xsmall"
+            icon={<IconRefresh />}
+          />
+        </Confirm>
+      </UrlContainer>
+    );
+  }
+}
+
+type Props = {
+  loading: boolean;
+  onToggle: () => void;
+  /**
+   * Called when refreshing an existing link
+   */
+  onReshare: () => void;
+  /**
+   * Link is public
+   */
+  isShared?: boolean;
+  shareUrl?: string | null;
+};
+
+class ShareIssue extends React.Component<Props> {
+  static propTypes = {
+    loading: PropTypes.bool.isRequired,
+    isShared: PropTypes.bool,
+    shareUrl: PropTypes.string,
+    onToggle: PropTypes.func.isRequired,
+    onReshare: PropTypes.func.isRequired,
+  };
+
+  hasConfirmModal = false;
+
+  handleToggleShare = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    this.props.onToggle();
+  };
+
+  handleOpen = () => {
+    const {loading, isShared, onToggle} = this.props;
+    if (!loading && !isShared) {
+      // Starts sharing as soon as dropdown is opened
+      onToggle();
+    }
+  };
+
+  // State of confirm modal so we can keep dropdown menu opn
+  handleConfirmCancel = () => (this.hasConfirmModal = false);
+  handleConfirmReshare = () => (this.hasConfirmModal = true);
```

Comment:
I think we should not return this 

```suggestion
  handleConfirmReshare = () => { this.hasConfirmModal = true };
```